### PR TITLE
unoconv: 0.6 -> 0.8.2

### PR DESCRIPTION
**Pull request by akirak**
###### Things done

Update unoconv, the document converter based on LibreOffice, from 0.6 to 0.8.2. 

unoconv 0.6 was released in 2012 --- It's outdated. It lacked some command line options I was desperate for, so I will send this PR.

The source repository was an HTTP site. I have changed it to GitHub.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/text/unoconv/default.nix [hunk 1]
```diff
@@ -1,4 +1,4 @@
-{ stdenv, fetchurl, python3, libreoffice-unwrapped, asciidoc, makeWrapper
+{ stdenv, pkgs, python3, libreoffice-unwrapped, asciidoc, makeWrapper
```
Comment (alyssais) [inline pkgs/tools/text/unoconv/default.nix]: Don’t take `pkgs` as an input, ever — every package could be accessed using it, but there are reasons we don’t do it and use `callPackage` instead. It’s difficult to override, and I believe it’s more efficient to use `callPackage`.

```suggestion
{ stdenv, fetchFromGitHub, python3, libreoffice-unwrapped, asciidoc, makeWrapper
```

File: pkgs/tools/text/unoconv/default.nix [hunk 2]
```diff
@@ -7,19 +7,19 @@
...
-    sha256 = "1m3kv942zf5rzyrbkil0nhmyq9mm3007y64bb3s7w88mhr5n23kr";
+  name = "unoconv";
+  version = "0.8.2";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "unoconv";
+    repo = "unoconv";
+    rev = "c3ff8937b12e5bf150861888ef86417f3f4de65c";
+    sha256 = "0ix605lk0k3hv241jb2kf4jq5744q2wh9x0pzkmay5m126vv8kq4";
+    # date = 2019-09-26T20:52:20+02:00;
```
Comment (alyssais) [inline pkgs/tools/text/unoconv/default.nix]: I don’t think we need this line.

File: pkgs/tools/text/unoconv/default.nix [hunk 3]
```diff
@@ -7,19 +7,19 @@
...
-  src = fetchurl {
-    url = "http://dag.wieers.com/home-made/unoconv/${name}.tar.gz";
-    sha256 = "1m3kv942zf5rzyrbkil0nhmyq9mm3007y64bb3s7w88mhr5n23kr";
+  name = "unoconv";
+  version = "0.8.2";
+
+  src = pkgs.fetchFromGitHub {
+    owner = "unoconv";
+    repo = "unoconv";
+    rev = "c3ff8937b12e5bf150861888ef86417f3f4de65c";
```
Comment (alyssais) [inline pkgs/tools/text/unoconv/default.nix]: Is there no tag for the release?

**Review by alyssais — CHANGES_REQUESTED** [review]

**Comment by akirak** [comment]
Thanks, I've fixed the issues.

File: pkgs/tools/text/unoconv/default.nix [hunk 4]
```diff
@@ -7,19 +7,18 @@
 # will not be able to load the pyuno module from libreoffice).
 
 stdenv.mkDerivation rec {
-  name = "unoconv-0.6";
-
-  src = fetchurl {
-    url = "http://dag.wieers.com/home-made/unoconv/${name}.tar.gz";
-    sha256 = "1m3kv942zf5rzyrbkil0nhmyq9mm3007y64bb3s7w88mhr5n23kr";
+  name = "unoconv";
```
Comment (bjornfor) [inline pkgs/tools/text/unoconv/default.nix]: Use "pname" instead of "name" here.

**Comment by akirak** [comment]
Fixed.

**Review by bjornfor — APPROVED** [review]

**Pull request merged**